ext/js/pages/settings/dictionary-controller.js

```diff
@@ -903,6 +1039,22 @@ export class DictionaryController {
         }
     }
 
+    /**
+     * @param {string} dictionaryTitle
+     */
+    async _updateDictionary(dictionaryTitle) {
+        if (this._checkingIntegrity || this._checkingUpdates || this._isDeleting || this._dictionaries === null) { return; }
+
+        const dictionaryInfo = this._dictionaries.find((entry) => entry.title === dictionaryTitle);
+        if (typeof dictionaryInfo === 'undefined') { throw new Error('Dictionary not found'); }
+        const {downloadUrl} = dictionaryInfo;
+        if (typeof downloadUrl !== 'string') { throw new Error('Attempted to update dictionary without download URL'); }
+
+        await this._deleteDictionary(dictionaryTitle);
```

Comment:
I suppose this is a failure case if `deleteDictionary` partially fails here. What are the mechanisms to maintain data consistency here?

Reply:
Worst case scenario for this feature is delete works, then import fails and you're left with no dict. Preventing that would be difficult.

Deleting itself hasn't been known to fail, I think.

Reply:
Deleting certainly can fail but for some reason it tends to be less common (maybe because you cant do it in bulk unless you delete all). One way around this would be storing dict metadata somewhere in the database and only deleting that after the import succeeds. So there could be an easy way to recover and maybe the user could press a button to start the import.